Basis/Packages/com.basis.framework/Avatar/BasisAvatarFactory.cs

```
@@ -9,304 +9,243 @@
+
+        public static async Task<GameObject> DownloadAndLoadAvatar(BasisLoadableBundle bundle, BasisPlayer player)
         {
-            if (Output.TryGetComponent(out BasisAvatar Avatar))
-            {
-                DeleteLastAvatar(Player);
-                Player.IsConsideredFallBackAvatar = false;
-                Player.BasisAvatar = Avatar;
-                Player.BasisAvatarTransform = Avatar.transform;
-                Player.BasisAvatar.Renders = Player.BasisAvatar.GetComponentsInChildren<Renderer>(true);
-                Player.BasisAvatar.IsOwnedLocally = Player.IsLocal;
-                switch (Player)
-                {
-                    case BasisLocalPlayer localPlayer:
-                        {
-                            SetupLocalAvatar(localPlayer);
-                            Avatar.OnAvatarReady?.Invoke(true);
-                            break;
-                        }
-                    case BasisRemotePlayer remotePlayer:
-                        {
-                            SetupRemoteAvatar(remotePlayer);
-                            Avatar.OnAvatarReady?.Invoke(false);
-                            break;
-                        }
-                }
+            string id = BasisGenerateUniqueID.GenerateUniqueID();
+            var obj = await BasisLoadHandler.LoadGameObjectBundle(bundle, true, player.ProgressReportAvatarLoad, new CancellationToken(), player.transform.position, Quaternion.identity, Vector3.one, false, BundledContentHolder.Selector.Avatar, player.transform);
+            player.ProgressReportAvatarLoad.ReportProgress(id, 100, "Setting Position");
+            obj.transform.SetPositionAndRotation(player.transform.position, Quaternion.identity);
+            return obj;
+        }
+
+        private static void InitializePlayerAvatar(BasisPlayer player, GameObject obj)
+        {
+            if (!obj.TryGetComponent(out BasisAvatar avatar)) return;
```

This will silently error if this ever occurs. leading to a bad avatar persisting